benchmarks/test_parser.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+import os
+import subprocess
+from pyquil._parser.parser import run_parser
+from pyquil.quil import Program
+
+
+def from_corpus():
+    programs = []
+    dir = os.path.join(os.path.dirname(__file__), "quilc", "tests", "good-test-files")
+    if not os.path.exists(dir):
+        subprocess.Popen(["git", "submodule", "update", "--init", "--recursive"]).wait()
```

Comment:
Does the `actions/checkout` in `bench.yml` ensure that the CI system will have `git` installed?

Reply:
You can definitely run it this way, but I'd probably prefer to use `subprocess.run` and `shlex.split` myself, and I'd probably make use of `pathlib` rather than dealing with the `os` module:

```python
from pathlib import Path
import shlex
import subprocess
from typing import List, Tuple

from lark import LarkError
from pyquil._parser.parser import run_parser


def from_corpus() -> List[Tuple[Path, str]]:
    programs = []
    DIR = Path(__file__).parent / "quilc" / "tests" / "good-test-files"
    if not DIR.exists():
        subprocess.run(shlex.split("git submodule update --init --recursive"))
    for path in DIR.glob("*.quil"):
        with path.open() as f:
            program = f.read()
            try:
                run_parser(program)
                programs.append((path, program))
            except LarkError:
                continue  # TODO log these or something?
    return programs
```

Reply:
awesome, thanks @genos. will update the PR with your recommendation!